code/datums/controllers/process/gang.dm

```diff
@@ -0,0 +1,161 @@
+			return //do nothing on our first run
+		else if (crate_spawn_repeats == 2)
+			schedule_interval = GANG_CRATE_DROP_FREQUENCY
+		var/turfList[0]
+		var/attempts = 0
+		var/area/drop_zone
+
+		while (attempts <= 3 && !length(turfList))
+			attempts++
+			drop_zone = pick(potential_hot_zones)
+			for (var/turf/simulated/floor/T in drop_zone.contents)
+				if (!is_blocked_turf(T))
+					turfList.Add(T)
+			if (!length(turfList))
+				logTheThing(LOG_DEBUG, null, "Couldn't find a valid location to drop a weapons crate inside [drop_zone.name].")
+		if (!length(turfList))
+			message_admins("All attempts to find a valid location to spawn a weapons crate failed!")
+			return
+		new/obj/storage/crate/gang_crate/guns_and_gear(pick(turfList))
+		broadcast_to_all_gangs("<span style='font-size:24px'> We've dropped off weapons & ammunition at <b>\the [drop_zone.name]!</b> It's anchored in place for 5 minutes, so get fortifying!</span>")
+
+
+		SPAWN(GANG_CRATE_LOCK_TIME - 1 MINUTE)
+			if(drop_zone != null)
+				broadcast_to_all_gangs("The weapons crate at the [drop_zone.name] can be moved in 1 minute!")
+			sleep(1 MINUTE)
+			if(drop_zone != null)
+				broadcast_to_all_gangs("The weapons crate at the [drop_zone.name] is free! Drag it to your locker.")
+
+/datum/controller/process/gang_duffle_drop
+	var/repeats = 1
+	var/duffle_spawn_repeats = 0
+	setup()
+		name = "Gang_Duffle_Drops"
+		schedule_interval = GANG_LOOT_INITIAL_DROP
+		src.repeats = GANG_LOOT_DROP_VOLUME_PER_GANG
+	doWork()
+		if (!istype(ticker.mode, /datum/game_mode/gang))
+			src.disable()
+			return
+		duffle_spawn_repeats++
+		if (duffle_spawn_repeats == 1)
+			return //do nothing on our first run
+		else if (duffle_spawn_repeats == 2)
+			schedule_interval = GANG_LOOT_DROP_FREQUENCY
+		var/datum/game_mode/gang/gamemode = ticker.mode
+		var/list/civiliansAlreadyPinged = list()// try not to have the same person picked twice
+		for(var/datum/gang/targetGang as anything in gamemode.gangs) //create loot bags for this gang (so they get pinged)
+			var/list/datum/mind/gangChosenCivvies = list() //which civilians have been picked for this gang
+			for(var/i = 1 to repeats)
+				var/datum/mind/civvie = targetGang.get_random_civvie(civiliansAlreadyPinged)
+				civiliansAlreadyPinged += civvie
+				if (!(civvie in gangChosenCivvies))
+					gangChosenCivvies += civvie
+				targetGang.target_loot_spawn(civvie)
+			var/broadcast_string = "<span style='font-size:20px'> Our associates have hidden [repeats] bag[s_es(repeats)] of weapons & supplies on board. The location[s_es(repeats)] have been tipped off to the PDAs of: "
+			if (length(gangChosenCivvies) > 1)
+				for (var/name=1 to length(gangChosenCivvies)-1)
+					broadcast_string += "[gangChosenCivvies[name].current.real_name] the [gangChosenCivvies[name].assigned_role]."
+				broadcast_string += "and [gangChosenCivvies[length(gangChosenCivvies)].current.real_name] the [gangChosenCivvies[length(gangChosenCivvies)].assigned_role]."
```

Comment:
```suggestion
				broadcast_string += " and [gangChosenCivvies[length(gangChosenCivvies)].current.real_name] the [gangChosenCivvies[length(gangChosenCivvies)].assigned_role]."
```
I'm not doing code review yet dw. Buuut actually there's a helper proc `english_list` for this kind of thing.